packages/flag-evaluation/src/index.ts

```diff
@@ -203,26 +203,44 @@ export interface Rule<T extends RuleValue> {
  * @return {Record<string, string>} A flattened JSON object with "stringified" keys and values.
  */
 export function flattenJSON(data: object): Record<string, string> {
-  if (Object.keys(data).length === 0) return {};
-  const result: Record<string, any> = {};
-  function recurse(cur: any, prop: string) {
-    if (Object(cur) !== cur) {
-      result[prop] = cur;
-    } else if (Array.isArray(cur)) {
-      const l = cur.length;
-      for (let i = 0; i < l; i++)
-        recurse(cur[i], prop ? prop + "." + i : "" + i);
-      if (l == 0) result[prop] = [];
+  const result: Record<string, string> = {};
+
+  if (Object.keys(data).length === 0) {
+    return result;
+  }
+
+  function recurse(value: any, prop: string = "") {
```

Comment:
Introducing default arguments for the heck if it just makes things more complicated IMO. Explicit is better than implicit.